cpp/daal/src/algorithms/tsne/tsne_gradient_descent_fpt.cpp

```diff
@@ -160,17 +161,19 @@ services::Status qTreeBuildingKernelImpl(IdxType * child, const DataType * posx,
     DataType r     = 0.;
     DataType px    = 0.;
     DataType py    = 0.;
+    const DataType localEps = 1e-6;
```

Comment:
What about using `std::numeric_limits<DataType>::epsilon()`. If you make it `constexpr ` that will guarantee that we do not depend on standard library.

```suggestion
    constexpr DataType localEps = std::numeric_limits<DataType>::epsilon();
```

Reply:
I didn't find similar uses of std in the old interfaces. Do you mind leaving it as it is?

Reply:
Okay. But such magical constants can vary dramatically for different types. That is probably not enough for let say `float16` type